server/controller/book.js

```diff
@@ -0,0 +1,344 @@
+import model from '../models';
+
+const bookModel = model.books;
+const borrowedBooks = model.borrowedbooks;
+const categoryModel = model.categories;
+
+
+/**
+ * @class Book
+ * @classdesc creates a Book classs
+ */
+class Book {
+  /**
+   * @param {object} req 
+   * @param {object} res
+   * @returns {void}
+   */
+  static addBook(req, res) {
+    bookModel.create(req.body).then((book) => {
+      res.status(201).json({ message: 'Book created', data: book });
+    }).catch((error) => {
+      res.status(501).json({ message: error.errors[0].message });
+      // check if all fields are supplied.
+      // if (error.name === 'SequelizeValidationError') {
+      //   res.status(400).json({ message: 'One or more fields are empty' });
+      //   // check if a duplicate request was made.
+      // } else if (error.name === 'SequelizeUniqueConstraintError') {
+      //   res.status(409).json({ message: 'A book with this ISBN already exists' });
+      // } else {
+      //   res.status(501).json({ error });
+      // }
+    });
+  }
+
+
+  /**
+   * @param {object} req 
+   * @param {object} res
+   * @returns {void}
+   */
+  static deleteBook(req, res) {
+    const bookId = parseInt((req.params.id), 10);
+    borrowedBooks.findOne({ where: { bookid: bookId, returnstatus: false } })
+      .then((response) => {
+        if (response !== null) {
+          res.status(501).json({ message: 'This book has been borrowed and cannot be deleted' });
+        } else {
+          bookModel.findOne({ where: { id: bookId, visibility: true } })
+            .then((reply) => {
+              if (reply === null) {
+                res.status(404).json({ message: 'Book not found in the database' });
+              } else if (reply.dataValues.visibility === true) {
+                bookModel.update({ visibility: false }, { where: { id: bookId } })
+                  .then(() => {
+                    bookModel.findAll().then((allbooks) => {
```

Comment:
Promise chaining can be used here to make your code more readable. Read about it [here](https://javascript.info/promise-chaining)